pyodide-build/pyodide_build/common.py

```diff
@@ -91,6 +93,49 @@ def find_matching_wheels(wheel_paths: Iterable[Path]) -> Iterator[Path]:
                 yield wheel_path
 
 
+def parse_top_level_import_name(whlfile: Path) -> list[str] | None:
+    """
+    Parse the top-level import name from a package file.
+    """
+
+    assert whlfile.name.endswith(".whl"), "Not a package file"
```

Comment:
Assert should not be used to check user input https://stackoverflow.com/a/17531188/1791279

Better to explicitly raise some exception if the conditions False.